41 emissions plots and outputs separate from concentration

rosypctals opened the pull request:
5-panel plot of total emissions for each pollutant!

lkoolik commented on supporting/concentration_layer.py:
```diff
@@ -231,6 +241,77 @@ def process_emissions(self, emis, isrm_obj, verbose, output_dir, output_emis_fla
             
         return tmp_dct['PM25'], tmp_dct['NH3'], tmp_dct['VOC'], tmp_dct['NOX'], tmp_dct['SOX']
     
+    def visualize_emissions(self, pollutant_name=''):
```
Can you add more comments to this new function? Each little code chunk should have a header, and if there's anything that isn't immediately intuitive, it should be noted

lkoolik commented on supporting/concentration_layer.py:
```diff
@@ -231,6 +241,77 @@ def process_emissions(self, emis, isrm_obj, verbose, output_dir, output_emis_fla
+            output_region.boundary.plot(ax=ax, edgecolor='black', facecolor = 'none')  
+            
+            # Add north arrow
+            ax.annotate('', xy=(0.94, 0.95), xytext=(0.94, 0.92), arrowprops=dict(facecolor='black', shrink=0.4),
```
Is this always north? I think the answer is probably yes because we're forcing it to the ISRM grid, right? Maybe just check the code pipeline to 100% confirm that this is always true. Also, does the ISRM's CRS actually orient with true north? Projections are hard!

rosypctals replied:
I've emailed Chirag and I've written code that calculates the angle for a north arrow given a custom CRS. I know that north isn't always north when using a custom CRS, but I just followed up asking if it's also the case for ISRM's CRS. If it isn't, should I have the code somewhere regardless? There's two methods: one for calculating the angle, and one for creating the rotated arrow!

lkoolik replied:
I think it's probably best practice to always calculate the angle and make sure it is true, true north. The two methods you described seem like a great approach!

lkoolik commented on supporting/concentration_layer.py:
```diff
@@ -231,6 +241,77 @@ def process_emissions(self, emis, isrm_obj, verbose, output_dir, output_emis_fla
+            
+            # Clip to output region if provided
+            if output_region is not None:
+                data = gpd.clip(data, output_region)
```
Is there any way to do this clip outside of the for-loop? The function is taking a long time, and I'm trying to identify a few ways to speed it up

rosypctals commented:
I've added comments to the function and edited concentration_layers and concentration so they calculate and add the north arrows! I've also changed it so that the border is only around California, not the output regionl. I tried different ways to see if we can clip it only once outside the for loop, but I haven't been able to get it to work, since we still have to clip the data each time to the output region- but I've tried taking some of the code out of the for loop that I can. The current fastest mine has run was around 2.3 minutes, and without clipping at all it took around 2.0 minutes, so I hope it won't be *too* big of an issue? I'll take another look for sure!
